### PR TITLE
duck 5.3.8.23611

### PR DESCRIPTION
**Pull request by dkocher**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/duck.rb [hunk 1]
```diff
@@ -17,16 +17,18 @@ class Duck < Formula
   depends_on :java => ["1.8+", :build]
   depends_on :xcode => :build
   depends_on "ant" => :build
```
Comment (fxcoudert) [inline Formula/duck.rb]: Is `ant` still needed?

File: Formula/duck.rb [hunk 2]
```diff
@@ -17,16 +17,18 @@ class Duck < Formula
...
+    system "mvn", "-DskipTests", "-Dgit.commitsCount=#{revision}", "--projects", "cli/osx", "--also-make", "package"
+    libexec.install Dir["cli/osx/target/duck.bundle/*"]
     bin.install_symlink "#{libexec}/Contents/MacOS/duck" => "duck"
   end
 
   test do
-    system "#{bin}/duck", "--download", Formula["wget"].stable.url, testpath/"test"
-    (testpath/"test").verify_checksum Formula["wget"].stable.checksum
+    system "#{bin}/duck", "--download", Formula["when"].stable.url, testpath/"test"
+    (testpath/"test").verify_checksum Formula["when"].stable.checksum
```
Comment (fxcoudert) [inline Formula/duck.rb]: Why change formula?

File: Formula/duck.rb (hunk 1 above)
Comment (dkocher) [inline Formula/duck.rb]: Ant is still required invoked from the Maven Build.

File: Formula/duck.rb (hunk 2 above)
Comment (dkocher) [inline Formula/duck.rb]: At some point the stable URL for `wget` was not accessible.

**Comment by fxcoudert** [comment]
@dkocher I'm not sure… can you amend your commit to put the `wget` URL back? this should trigger a new build, and we'll see.

**Comment by dkocher** [comment]
Any idea why the pull from the bot is failing?

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by dkocher** [comment]
I will look into the test failure. One should add a checklist item
- [ ] Does your build pass `brew test <formula>` (after doing `brew install <formula>`)?

**Comment by dkocher** [comment]
```
==> brew audit duck --online
==> FAILED
Error: 1 problem in 1 formula
duck:
  * HEAD: The URL https://svn.cyberduck.io/trunk/ is not a valid svn URL
```

Could this be an issue with validating the server TLS certificates? I do not see the issue here on my system.

**Comment by JCount** [comment]
The audit message about `svn` is bogus in this instance. The connection/sever/planets were simply misaligned.

**Comment by fxcoudert** [comment]
@JCount no it's not a random network issue in that case. As @ilovezfs made me aware, pre-Sierra `svn` does not accept certain valid `https` certificates — which in turn typically gives this audit error, but only on Yosemite & El Capitan.

**Comment by JCount** [comment]
That's weird because I know of that issue so I tested on a Yosemite VM. I must have brewed svn installed without realizing it. Good catch.

**Comment by MikeMcQuaid** [comment]
I've opened https://github.com/Homebrew/brew/pull/2262 to address the Subversion audit issue.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
```
==> /usr/local/Cellar/duck/5.3.8.23611/bin/duck --download https://ftpmirror.gnu.org/wget/wget-1.19.1.tar.gz /tmp/duck-test-20170315-65244-6736p6/test
Error: Could not find or load main class ch.cyberduck.cli.ApplicationTerminal
Error: duck: failed
Failed executing: /usr/local/Cellar/duck/5.3.8.23611/bin/duck --download https://ftpmirror.gnu.org/wget/wget-1.19.1.tar.gz /tmp/duck-test-20170315-65244-6736p6/test
```

**Comment by fxcoudert** [comment]
Same failure with 5.4.0.23761. @dkocher could you report it upstream?

**Comment by dkocher** [comment]
I will review the failure.

**Comment by dkocher** [comment]
The difference that will lead the application to fail opening when installed with `brew` is an invalid rewrite of a dynamic library dependency. The dependency on `libjvm.dylib` is rewritten to a system installation.

```diff
18c18
< 	@executable_path/../PlugIns/Runtime.jre/Contents/Home/lib/server/libjvm.dylib (compatibility version 1.0.0, current version 1.0.0)
---
> 	/System/Library/Frameworks/JavaVM.framework/Versions/A/JavaVM (compatibility version 1.0.0, current version 1.0.0)
```
Can we disable this rewrite of the dependencies on installation?

**Comment by dkocher** [comment]
The dynamic library dependency should now be fixed with the build goal change. What is the reason of the current build failure?

**Comment by JCount** [comment]
@BrewTestBot test this please

**Comment by JCount** [comment]
@BrewTestBot test this please

**Comment by JCount** [comment]
@dkocher Only the 'default' CI matters, the other failure is a false positive. Does it make any sense to update this PR to 5.4.4 before merging? I see that it is now the latest stable release.

**Comment by dkocher** [comment]
I will open a new PR with the latest bugfix update.

**Comment by JCount** [comment]
@BrewTestBot test this please

**Comment by JCount** [comment]
@dkocher Thank you for your contribution to Homebrew! 😄 

I decided to go ahead and merge this, but feel free to open a new PR to updated to the most recent stable if you would like. 👍

**Pull request closed without merging**